### PR TITLE
Measure timer, improve fairness

### PR DESCRIPTION
**Pull request by pdobacz**
Two changes thrown in together:
1/ make the measurement cover the entire interpreter loop and not print to stdout in the loop
2/ measure the timer as well

Related to imapp-pl/gas-cost-estimator#35  imapp-pl/gas-cost-estimator#38

Excuse the crudeness, we can work on making elegant later on. Also we don't want any additional function calls etc.

### CODE REVIEW

File: ethcore/evm/src/interpreter/mod.rs [hunk 1]
```diff
@@ -222,23 +222,69 @@ impl<Cost: 'static + CostType> vm::Exec for Interpreter<Cost> {
...
+                            opcode_duration = timer.elapsed().as_nanos();
+                            timer = howlong::timer::SteadyTimer::new();
+                            timer_duration = timer.elapsed().as_nanos();
+
+                            timer_durations.push(timer_duration);
+                            opcode_durations.push(opcode_duration);
+
+                            for n in 0..timer_durations.len() {
+                                println!("{:?},{:?},{:?},{:?}", sample, n, opcode_durations[n], timer_durations[n]);
+                            }
```
Comment (pik694) [inline ethcore/evm/src/interpreter/mod.rs:274]: Can we lift it to a func? Maybe in this code we ensure that there is a little additional calls as possible, but the code is unfortunately copy-pasted in 4 places, which seems reasonable to extract it to become a func.

**Review by pik694 — APPROVED** [review]

File: ethcore/evm/src/interpreter/mod.rs (hunk 1 above)
Comment (pdobacz) [reply ethcore/evm/src/interpreter/mod.rs:274]: @pik694 :blush: I'd need to learn more rust... it takes several minutes to build this thing... do we have to :) ?

I'll give it a try, but time cap it to say 10 mins. If you know how to, pls lmk and just commit the change

Comment (pik694) [reply ethcore/evm/src/interpreter/mod.rs:274]: You can leave it to me then 😉

Comment (pdobacz) [reply ethcore/evm/src/interpreter/mod.rs:274]: oh @pik694 but please don't move the capturing of time to the function, only the pieces that are "outside" of the measured execution!

The 
```rust
                             opcode_duration = timer.elapsed().as_nanos();
                            timer = howlong::timer::SteadyTimer::new();
                            timer_duration = timer.elapsed().as_nanos(); 
```

should stay where it is :(

Comment (pik694) [reply ethcore/evm/src/interpreter/mod.rs:274]: we could just use an inlined function like this:
```rust
#[inline(always)]
fn measure(
    timer: &mut howlong::timer::SteadyTimer,
    opcode_duration: &mut u128,
    timer_duration: &mut u128,
    timer_durations: &mut Vec<u128>,
    opcode_durations: &mut Vec<u128>,
) {
    *opcode_duration = timer.elapsed().as_nanos();
    *timer = howlong::timer::SteadyTimer::new();
    *timer_duration = timer.elapsed().as_nanos();

    timer_durations.push(*timer_duration);
    opcode_durations.push(*opcode_duration);
}
```

But I'm not quite sure if we should do it and hope that the compiler would care to listen to our hints - though the doc says it should: https://nnethercote.github.io/perf-book/inlining.html

Comment (pdobacz) [reply ethcore/evm/src/interpreter/mod.rs:274]: if in doubt leave it out :). If we ever need to modify this code in the future (I don't think so) we can fix this.

**Pull request merged**